Gem/Code/Source/AutoGen/NetworkMatchComponent.AutoComponent.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+
+<Component
+    Name="NetworkMatchComponent" 
+    Namespace="MultiplayerSample" 
+    OverrideComponent="false" 
+    OverrideController="true"
+	OverrideInclude="Source/Components/NetworkMatchComponent.h"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+
+    <NetworkProperty Type="float" Name="RoundTime" Init="120.0f" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="true" IsPredictable="false" ExposeToEditor="false" ExposeToScript="false" GenerateEventBindings="true" Description="The remaining time in the round" />
```

Comment:
Just as a heads up, this is a pretty classic mistake for new to multiplayer folks. A float necessarily serializes as 4-bytes, and the way this is set up that float will change every tick. That means we are forced to send that whole 4-byte value every tick.

The alternative approach is to use a quantized or integral representation of time, and don't put the actual current timer value into the network property, instead put the end value you're counting down to (or up to).

As an example, say I have a cooldown on an invulnerability buff. Rather than have a float buffTimeRemaining network property, instead you'd probably want an AzTimeMs buffTimeEndMs or something similar. Then your client can take buffTimeEndMs() - GameCurrentTimeMs() to reconstruct the timer value, and you transmit the buff time once rather than every frame.

I don't know that any changes are required here since this is probably a level entity, but hopefully it's helpful for any further multiplayer component work we do on the new sample.

Also recommend always putting units onto time values, especially if you're not using type-safe representations of time. I don't know if RoundTime is in seconds, milliseconds, nanoseconds, kiloseconds, etc.. If you check the rest of the networking and multiplayer codebases we're quite strict on using type-safe representations of time, and putting the units into the variable names itself just so there's no customer confusion or we have someone trying to add milliseconds onto a second timer without realizing it.

Reply:
Good call, I am addressing that in https://github.com/aws-lumberyard-dev/o3de-multiplayersample/pull/18